src/pages/CustomerInvoicePage.js

```diff
@@ -68,10 +66,6 @@ export const CustomerInvoice = ({
 }) => {
   const { isFinalised, comment, theirRef } = pageObject;
 
-  // Listen for this invoice being finalised which will prune items and cause side effects
-  // outside of the reducer. Reconcile differences when triggered.
-  useRecordListener(refreshData, pageObject, 'Transaction');
```

Comment:
removed these `recordListener`s in the finalisable pages - the `PagesReducer` is hooked into the `finalise` action instead?